src/mumble/JackAudio.cpp

```diff
@@ -1059,6 +1125,92 @@ bool JackAudioOutput::process(const jack_nframes_t frames) {
 	return true;
 }
 
+void JackAudioOutput::prepareOutputBuffers(unsigned int frameCount, QList< AudioOutputBuffer * > &qlMix,
+										   QList< AudioOutputBuffer * > &qlDel) {
+	ServerHandlerPtr sh = Global::get().sh;
+	VoiceRecorderPtr recorder;
+	if (sh) {
+		recorder = Global::get().sh->recorder;
+
+		// Shortcut to base class code when our special case is not needed.
+		if (recorder && !recorder->isTransportEnabled()) {
+			AudioOutput::prepareOutputBuffers(frameCount, qlMix, qlDel);
+			return;
+		}
+	}
+
+	// Register user ports based on who is currently present in qmOutputs and route their audio to JACK.
+	// Don't care if users get removed for now. TODO: maybe at some point we do.
+	auto it = qmOutputs.constBegin();
+	while (it != qmOutputs.constEnd()) {
```

Comment:
```suggestion
	for (auto it = qmOutputs.constBegin(); it != qmOutputs.constEnd(); ++it) {
```
(and remove the  `it++` from the loop body)
There are no circumstances in which you want to enter a new loop iteration but don't want to increment `it`, correct?

Reply:
Correct!